### PR TITLE
Adding SSL to v3 monitoring

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

File: docker/oso-rhel7-zagg-web/root/config.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@
     g_zagg_server_config: "{{ lookup('env','ZAGG_SERVER_CONFIG') | default(g_default_zagg_server_config, True) }}"
     g_zagg_server_user: "{{ lookup('env','ZAGG_SERVER_USER') | default(g_default_zagg_server_user, True) }}"
     g_zagg_server_password: "{{ lookup('env','ZAGG_SERVER_PASSWORD') | default(g_default_zagg_server_password, True) }}"
+    g_zagg_ssl_verify: "{{ lookup('env','ZAGG_SSL_VERIFY') | default(g_default_zagg_ssl_verify, True) }}"
```
Comment (twiest) [inline docker/oso-rhel7-zagg-web/root/config.yml]: Recommend removing "True" from the default filter call.

File: docker/oso-rhel7-zagg-web/root/default_vars.yml [hunk 2]
```diff
@@ -5,10 +5,13 @@ g_default_zagg_server_config:
...
-    api_url: http://oso-rhel7-zabbix-web/zabbix/api_jsonrpc.php
+    api_url: https://oso-rhel7-zabbix-web/zabbix/api_jsonrpc.php
     api_user: apiuser
     api_password: apipass
+    ssl_verify: False
     path: /var/run/zagg/data/local-cluster-zbx
 
 g_default_zagg_server_user: zagg-user
 g_default_zagg_server_password: zagg-password
+g_default_zagg_ssl_verify: False
```
Comment (twiest) [inline docker/oso-rhel7-zagg-web/root/default_vars.yml]: remove this

File: docker/oso-rhel7-zagg-web/run.sh [hunk 3]
```diff
@@ -22,7 +22,7 @@
 #
 
 
-export ZAGG_SERVER_CONFIG='{ "targets": [ { "name": "local zaio", "type": "zabbix", "trapper_server": "oso-rhel7-zaio", "trapper_port": 10051, "api_url": "http://oso-rhel7-zaio/zabbix/api_jsonrpc.php", "api_user": "Admin", "api_password": "zabbix", "path": "/var/run/zagg/data/local_zaio", }, ] }'
+export ZAGG_SERVER_CONFIG='{ "targets": [ { "name": "local zaio", "type": "zabbix", "trapper_server": "oso-rhel7-zaio", "trapper_port": 10051, "api_url": "https://oso-rhel7-zaio/zabbix/api_jsonrpc.php", "api_user": "Admin", "api_password": "zabbix", "ssl_verify": False, "path": "/var/run/zagg/data/local_zaio", }, ] }'
```
Comment (twiest) [inline docker/oso-rhel7-zagg-web/run.sh]: Sync this up with the comment above as that's what's used for editing.

File: openshift_tools/web/rest.py [hunk 4]
```diff
@@ -38,22 +55,29 @@ def _auth(self):
...
+            requests.packages.urllib3.disable_warnings()
+
         _headers = self.headers or {}
 
         if headers:
             _headers.update(headers)
 
         response = requests.request(
-            auth=None if not self._auth else self._auth,
+            auth=None if not self._auth else self._auth, allow_redirects=True,
```
Comment (twiest) [inline openshift_tools/web/rest.py]: consider putting allow_redirects on a separate line.

File: scripts/monitoring/ops-zagg-client.py [hunk 5]
```diff
@@ -82,16 +83,26 @@ def parse_config(self, config_file):
...
+        zagg_url = self.args.zagg_url if self.args.zagg_url else self.config['zagg']['url']
         zagg_user = self.args.zagg_user if self.args.zagg_user else self.config['zagg']['user']
         zagg_password = self.args.zagg_pass if self.args.zagg_pass else self.config['zagg']['pass']
-        host = self.args.host if self.args.host else self.config['host']['name']
 
-        zagg_conn = ZaggConnection(host=zagg_server,
+        zagg_ssl_verify = self.config['zagg'].get('ssl_verify', False)
+
+        if isinstance(zagg_ssl_verify, str):
+            zagg_ssl_verify = zagg_ssl_verify == 'True'
```
Comment (twiest) [inline scripts/monitoring/ops-zagg-client.py]: is this being used?

File: scripts/monitoring/ops-zagg-client.py [hunk 6]
```diff
@@ -82,16 +83,26 @@ def parse_config(self, config_file):
...
+
+        if self.args.zagg_ssl_verify:
+            zagg_ssl_verify = self.args.zagg_pass
```
Comment (twiest) [inline scripts/monitoring/ops-zagg-client.py]: this looks wrong.

File: scripts/monitoring/ops-zagg-client.py (hunk 5 above)
Comment (twiest) [inline scripts/monitoring/ops-zagg-client.py]: consider adding parenthesis like in ops-zagg-processor

File: docker/oso-rhel7-zagg-web/run.sh (hunk 3 above)
Comment (kwoodson) [inline docker/oso-rhel7-zagg-web/run.sh]: Done.

File: docker/oso-rhel7-zagg-web/run.sh [hunk 7]
```diff
@@ -15,6 +15,7 @@
 #      "api_url": "http://oso-rhel7-zaio/zabbix/api_jsonrpc.php",
```
Comment (twiest) [inline docker/oso-rhel7-zagg-web/run.sh]: https

**Comment by twiest** [comment]
:+1:

**Pull request merged**